### ISSUE TITLE
Catalog templates are not available in airgapped setup for at least 10 minutes

### ISSUE THREAD

**Issue by galal-hussein** [rancher/rancher#9669]
**Rancher versions:**
rancher/server: v1.6.8-rc1

**Steps to Reproduce:**

- Start Rancher server in airgap setup with no internet access

**Results:**
In infrastructure tab the infrastructure stacks are not showing, and i can see any environment templates when trying to add environment.

rancher server logs:
```
time="2017-08-11T22:30:19Z" level=info msg="Creating schema host, roles [admin user readAdmin]" id=1ds17 service=gms 
time="2017-08-11T22:30:19Z" level=info msg="Creating schema machine, roles [readonly]" id=1ds18 service=gms 
time="2017-08-11T22:30:20Z" level=info msg="Creating schema host, roles [readonly]" id=1ds19 service=gms 
time="2017-08-11T22:30:20Z" level=info msg="Updating machine jsons for  [digitalocean packet amazonec2 azure digitalocean]" service=gms 
time="2017-08-11T22:30:20Z" level=info msg="Creating schema machine, roles [service]" id=1ds20 service=gms 
time="2017-08-11T22:30:21Z" level=info msg="Creating schema host, roles [service]" id=1ds21 service=gms 
time="2017-08-11T22:30:21Z" level=info msg="Creating schema machine, roles [project member owner]" id=1ds22 service=gms 
time="2017-08-11T22:30:21Z" level=info msg="Creating schema host, roles [project member owner]" id=1ds23 service=gms 
time="2017-08-11T22:30:21Z" level=info msg="Creating schema machine, roles [admin user readAdmin]" id=1ds24 service=gms 
time="2017-08-11T22:30:22Z" level=info msg="Creating schema host, roles [admin user readAdmin]" id=1ds25 service=gms 
time="2017-08-11T22:30:22Z" level=info msg="Creating schema machine, roles [readonly]" id=1ds26 service=gms 
time="2017-08-11T22:30:22Z" level=info msg="Creating schema host, roles [readonly]" id=1ds27 service=gms 
time="2017-08-11T22:30:44Z" level=error msg="Error processing event" err="Get https://github.com/packethost/docker-machine-driver-packet/releases/download/v0.1.2/docker-machine-driver-packet_linux-amd64.zip: dial tcp 192.30.253.112:443: i/o timeout" eventId=bbbfb6c7-327d-4cab-8003-c6748c42e274 eventName="machinedriver.activate;handler=goMachineService-machine" resourceId=1md1 
2017-08-11 22:30:44,824 ERROR [8f5b4a1b-0946-4192-8817-56061338ba3d:14] [machineDriver:1] [machinedriver.activate] [] [ecutorService-1] [c.p.e.p.i.DefaultProcessInstanceImpl] Agent error for [machinedriver.activate.reply;handler=goMachineService-machine]: Get https://github.com/packethost/docker-machine-driver-packet/releases/download/v0.1.2/docker-machine-driver-packet_linux-amd64.zip: dial tcp 192.30.253.112:443: i/o timeout 
2017-08-11 22:30:44,841 ERROR [:] [] [] [] [ecutorService-1] [.e.s.i.ProcessInstanceDispatcherImpl] Agent error for [machinedriver.activate.reply;handler=goMachineService-machine]: Get https://github.com/packethost/docker-machine-driver-packet/releases/download/v0.1.2/docker-machine-driver-packet_linux-amd64.zip: dial tcp 192.30.253.112:443: i/o timeout 
time="2017-08-11T22:30:44Z" level=info msg=Event eventId=289c2693-65de-4ac6-a85f-6db421fc594b name="machinedriver.error;handler=goMachineService-machine" resourceId=1md1 service=gms 
time="2017-08-11T22:34:58Z" level=error msg="Failed to update existing repo cache: exit status 128" 
time="2017-08-11T22:34:59Z" level=error msg="Failed to update existing repo cache: exit status 128" 
time="2017-08-11T22:35:06Z" level=error msg="Failed to update existing repo cache: exit status 128" 
time="2017-08-11T22:39:46Z" level=error msg="Failed to update existing repo cache: exit status 128" 
time="2017-08-11T22:39:46Z" level=error msg="Failed to update existing repo cache: exit status 128" 
time="2017-08-11T22:39:46Z" level=info msg="Stack Create Event Received" eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:46Z" level=info msg="Stack Create Event Received" eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:46Z" level=info msg="Stack Create Event Received" eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:46Z" level=info msg="Stack Create Event Received" eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:46Z" level=info msg="[scheduler:]: Creating project " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:46Z" level=info msg="[scheduler:scheduler]: Creating " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:46Z" level=info msg="Creating service scheduler" 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:]: Creating project " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:network-manager]: Creating " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:metadata]: Creating " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:]: Creating project " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:healthcheck]: Creating " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="Creating service healthcheck" 
time="2017-08-11T22:39:47Z" level=info msg="Creating service metadata" 
time="2017-08-11T22:39:47Z" level=info msg="Creating service network-manager" 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:]: Creating project " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:cni-driver]: Creating " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:ipsec]: Creating " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="Creating service ipsec" 
time="2017-08-11T22:39:47Z" level=info msg="Creating service cni-driver" 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:metadata]: Created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:dns]: Creating " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:dns]: Created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:network-manager]: Created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:]: Project created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:]: Creating project " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:network-manager]: Creating " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:metadata]: Creating " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:healthcheck]: Created " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:]: Project created " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:]: Creating project " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:healthcheck]: Creating " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[scheduler:scheduler]: Created " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:47Z" level=info msg="[scheduler:]: Project created " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:47Z" level=info msg="[scheduler:]: Creating project " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:47Z" level=info msg="[scheduler:scheduler]: Creating " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:network-manager]: Created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:healthcheck]: Created " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[healthcheck:]: Project created " eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:ipsec]: Created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:router]: Creating " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:metadata]: Created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:dns]: Creating " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:router]: Created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:dns]: Created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[network-services:]: Project created " eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="Stack Create Event Done" eventId=0ca755f6-793c-4e9e-8d81-887dddcac4ea resourceId=1st3 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:cni-driver]: Created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:]: Project created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:]: Creating project " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="Stack Create Event Done" eventId=a99fd208-696e-4b71-81c5-7928610c0313 resourceId=1st1 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:cni-driver]: Creating " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:ipsec]: Creating " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[scheduler:scheduler]: Created " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:47Z" level=info msg="[scheduler:]: Project created " eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:47Z" level=info msg="Stack Create Event Done" eventId=675d40b5-9f68-419e-9941-e61a4ff8be1d resourceId=1st4 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:cni-driver]: Created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:ipsec]: Created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:router]: Creating " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:router]: Created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="[ipsec:]: Project created " eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:47Z" level=info msg="Stack Create Event Done" eventId=d38be48b-ea78-4bc3-a204-e804e0739a9a resourceId=1st2 
time="2017-08-11T22:39:53Z" level=error msg="Failed to update existing repo cache: exit status 128"
```

**Comment by deniseschannon** [comment]
Available in master of v1.6 with catalog-service v0.19.8. Will be available in v1.6.8-rc6

**Comment by galal-hussein** [comment]
Rancher version: v1.6.8-rc6
airgapped setup

I can see all infra stacks being deployed in the default environment with no delay and can see all environment templates too.
I can see the logs skipping the catalog refresh:
```
2017-08-25 19:40:19,896 WARN  [:] [] [] [] [ecutorService-3] [i.c.p.s.c.impl.CatalogServiceImpl   ] Skipping catalog refresh because it failed. Response code: 500, reason: Internal Server Error 
```
